subprojects/docs/src/docs/userguide/writing_build_scripts.adoc

```diff
@@ -119,37 +125,47 @@ There are two kinds of variables that can be declared in a build script: local v
 
-=== Example: Using local variables
+[.multi-language-text.lang-kotlin]
+Local variables are declared with the `val` keyword. They are only visible in the scope where they have been declared. Local variables are a feature of the underlying Kotlin language.
 
-[source.multi-language-sample,groovy]
-.build.gradle
-----
-include::{samplesPath}/userguide/tutorial/localVariables/build.gradle[]
-----
+.Using local variables
+====
+include::sample[dir="userguide/tutorial/localVariables/groovy",files="build.gradle[]"]
+include::sample[dir="userguide/tutorial/localVariables/kotlin",files="build.gradle.kts[]"]
+====
 
 [[sec:extra_properties]]
 === Extra properties
 
-All enhanced objects in Gradle's domain model can hold extra user-defined properties. This includes, but is not limited to, projects, tasks, and source sets. Extra properties can be added, read and set via the owning object's `ext` property. Alternatively, an `ext` block can be used to add multiple properties at once.
+All enhanced objects in Gradle's domain model can hold extra user-defined properties. This includes, but is not limited to, projects, tasks, and source sets.
 
-=== Example: Using extra properties
+[.multi-language-text.lang-groovy]
+Extra properties can be added, read and set via the owning object's `ext` property. Alternatively, an `ext` block can be used to add multiple properties at once.
 
-[source.multi-language-sample,groovy]
-.build.gradle
-----
-include::{samplesPath}/userguide/tutorial/extraProperties/groovy/build.gradle[tag=extraProperties]
-----
+[.multi-language-text.lang-kotlin]
+Extra properties can be added, read and set via the owning object's `extra` property. Alternatively, they can be addressed via Kotlin delegated properties using `by extra`.
+
+.Using extra properties
+====
+include::sample[dir="userguide/tutorial/extraProperties/groovy",files="build.gradle[tags=extraProperties]"]
+include::sample[dir="userguide/tutorial/extraProperties/kotlin",files="build.gradle.kts[tags=extraProperties]"]
 
 .Output of **`gradle -q printProperties`**
 ----
 > gradle -q printProperties
 include::{samplesPath}/userguide/tutorial/extraProperties/extraProperties.out[]
 ----
+====
 
+[.multi-language-text.lang-groovy]
 In this example, an `ext` block adds two extra properties to the `project` object. Additionally, a property named `purpose` is added to each source set by setting `ext.purpose` to `null` (`null` is a permissible value). Once the properties have been added, they can be read and set like predefined properties.
 
+[.multi-language-text.lang-kotlin]
+In this example, two extra properties are added to the `project` object using `by extra`. Additionally, a property named `purpose` is added to each source set by setting `extra["purpose"]` to `null` (`null` is a permissible value). Once the properties have been added, they can be read and set like predefined properties.
```

Comment:
> they can be read and set like predefined properties

Is that true? I understand this sentence as "if an extra property "foo" has been added to a project's extra properties, you can use `project.foo` to read or write it.". That's true in Groovy, but not in Kotlin, right?

Reply:
Absolutely right! Fixed, thanks for spotting this!